perun-core/src/main/java/cz/metacentrum/perun/core/impl/FacilitiesManagerImpl.java

```diff
@@ -336,6 +342,18 @@ public Facility getFacilityByName(PerunSession sess, String name) throws Facilit
 		}
 	}
 
+	@Override
+	public List<Facility> getFacilitiesByIds(PerunSession perunSession, List<Integer> ids) {
+		try {
+			SqlParameterSource parameters = new MapSqlParameterSource("ids", ids);
+
+			return namedParameterJdbcTemplate.query("select " + facilityMappingSelectQuery + " from facilities where id in (:ids)",
```

Comment:
Using ``id in (...)`` is not effective for large number of IDs. Please see ``UsersManagerImpl.getUsersByIds()`` implementation. This apply to impl of all methods.

Reply:
Done